src/NServiceBus.Core/Performance/TimeToBeReceived/TimeToBeReceived.cs

```diff
@@ -31,7 +31,7 @@ static TimeToBeReceivedMappings GetMappings(FeatureConfigurationContext context)
                 convention = userDefinedConvention.GetTimeToBeReceivedForMessage;
             }
 
-            var doesTransportSupportDiscardIfNotReceivedBefore = context.Settings.DoesTransportSupportConstraint<DiscardIfNotReceivedBefore>();
+            var doesTransportSupportDiscardIfNotReceivedBefore = context.Settings.Get<TransportDefinition>().SupportsTTBR;
```

Comment:
we could add TransportDefinition to the feature context, and even make that publicly accessible if we want to allow transport re-configuration at feature setup time.